packages/shuttle/src/shuttle/hubSubscriber.ts

```
@@ -152,9 +153,18 @@ export class BaseHubSubscriber extends HubSubscriber {
       }
 
       try {
+        // Do not allow hanging unresponsive connections to linger:
+        let cancel = setTimeout(() => {
+          this.destroy();
```

@CassOnMars doesn't this timeout callback need to throw an Error or do a `this.emit("onError")` after calling `this.destroy()`? Else how will the Shuttle-using app get notified of the need to exit/restart?

---

It closes out the connection and terminates the awaiting loop, producing a connection cancellation error:
```
Hub event stream processing halted unexpectedly: 1 CANCELLED: Call cancelled
HubSubscriber shuttle restarting hub event stream in 5 seconds...
```